### PR TITLE
0.1.9 rc with p2p changes

### LINKED ISSUE #2074: P2P device pairing is unstable
- [ ] The receiving devices already shows up on the sending device even before the receiving device's *Receive* button is clicked
- [ ] If you try to connect to the receiving device from the sending device before the *Receive* button on the receiving device is clicked, the connection fails.
- [ ] If you try to connect to the receiving device before the *Receive* button is clicked, the connection fails and subsequent connections also fail. After this, connecting both devices only works by reopenning the activity
- [ ] If you try to connect to the receiving device from the sending device just after clicking the *Receive* button, the connection also fails

### PR DESCRIPTION
**Pull request by ekigamba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2074 #2072 #2073 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by dubdabasoduba — DISMISSED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/p2p/dao/BaseP2PTransferDao.kt [hunk 1]
```diff
@@ -92,42 +92,44 @@ constructor(
...
+        )
 
-      /*  val searchQuery =
+        sort(DateClientParam(SyncDataParams.LAST_UPDATED_KEY), Order.ASCENDING)
+        from = offset
+        count = batchSize
+      }
+      fhirEngine.search(search)
+       */
+
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/p2p/dao/BaseP2PTransferDao.kt:111]: can we remove all these lines?

File: android/engine/src/test/java/org/smartregister/fhircore/engine/p2p/dao/BaseP2PTransferDaoTest.kt [hunk 2]
```diff
@@ -156,17 +167,20 @@ class BaseP2PTransferDaoTest : RobolectricTest() {
...
+    // Uncomment this when use of custom queries is no longer required
+    /*    assertEquals(ResourceType.Patient, searchSlot.captured.type)
 
     val dateTimeFilterCriterion: MutableList<Any> =
       ReflectionHelpers.getField(searchSlot.captured, "dateTimeFilterCriteria")
     val tokenFilters: MutableList<DateParamFilterCriterion> =
       ReflectionHelpers.getField(dateTimeFilterCriterion[0], "filters")
     assertEquals("_lastUpdated", tokenFilters[0].parameter.paramName)
-    assertEquals(ParamPrefixEnum.GREATERTHAN_OR_EQUALS, tokenFilters[0].prefix)
+    assertEquals(ParamPrefixEnum.GREATERTHAN_OR_EQUALS, tokenFilters[0].prefix)*/
```
Comment (pld) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/p2p/dao/BaseP2PTransferDaoTest.kt:183]: I think we should delete, put this in an issue

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainActivity.kt [hunk 3]
```diff
@@ -120,6 +127,19 @@ open class AppMainActivity : BaseMultiLanguageActivity(), QuestionnaireHandler,
...
+    appMainViewModel.viewModelScope.launch(dispatcherProvider.io()) {
+      try {
+        val addDateTimeIndexEntityIndexFromIndexQuery =
+          SearchQuery(
+            "CREATE INDEX `index_DateTimeIndexEntity_index_from` ON `DateTimeIndexEntity` (`index_from`)",
+            emptyList()
+          )
+        fhirEngine.search<Task>(addDateTimeIndexEntityIndexFromIndexQuery)
+      } catch (ex: SQLException) {
+        Timber.e(ex)
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainActivity.kt:140]: let's add a test that throws this

**Review by pld — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/p2p/dao/BaseP2PTransferDao.kt (hunk 1 above)
Comment (Rkareko) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/p2p/dao/BaseP2PTransferDao.kt]: Yes. Updated

**Review by pld — APPROVED** [review]

**Pull request merged**